### PR TITLE
feat: Set comment options on hover and display message deleted

### PR DESCRIPTION
**Pull request by fredrikleknes**

### CODE REVIEW

File: src/Components/SideSheet/Comments/Components/RenderComment.tsx [hunk 1]
```diff
@@ -52,18 +61,67 @@ const updateComment = async (
     }
 }
 
+const deleteComment = async (
+    softDelete: boolean,
+    comment: ReviewComment,
+    reviewComments: ReviewComment[],
+    setReviewComments: Dispatch<SetStateAction<ReviewComment[]>>,
+) => {
+    if (comment.id && comment.text) {
```
Comment (taustad) [inline src/Components/SideSheet/Comments/Components/RenderComment.tsx:70]: "" is falsy in JS, so you are not able to delete an empty string

File: src/Components/SideSheet/Comments/Components/RenderComment.tsx [hunk 2]
```diff
@@ -52,18 +61,67 @@ const updateComment = async (
...
+        if (softDelete) {
```
Comment (taustad) [inline src/Components/SideSheet/Comments/Components/RenderComment.tsx:71]: Is this function ever called with softDelete = false?

Comment (fredrikleknes) [reply src/Components/SideSheet/Comments/Components/RenderComment.tsx:71]: No, this was for if we ever needed to hard delete comments in the future

Comment (taustad) [reply src/Components/SideSheet/Comments/Components/RenderComment.tsx:71]: I think we should remove code that serves no purpose unless it is immediately needed in the future

File: src/Components/SideSheet/Comments/Components/RenderComment.tsx (hunk 1 above)
Comment (fredrikleknes) [reply src/Components/SideSheet/Comments/Components/RenderComment.tsx]: removed

File: src/Components/SideSheet/Comments/Components/RenderComment.tsx (hunk 2 above)
Comment (fredrikleknes) [reply src/Components/SideSheet/Comments/Components/RenderComment.tsx]: removed

File: src/api/BaseService.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ export class BaseService {
     }
 
     private async request(path: string, options?: RequestOptions): Promise<any> {
-        const { data } = await this.client.request({
+        const { data, headers } = await this.client.request({
```
Comment (taustad) [inline src/api/BaseService.ts]: Does this do anything?

File: src/Components/SideSheet/Comments/Components/RenderComment.tsx [hunk 4]
```diff
@@ -52,10 +48,34 @@ const updateComment = async (
...
 
+const deleteComment = async (
+    comment: ReviewComment,
+    reviewComments: ReviewComment[],
+    setReviewComments: Dispatch<SetStateAction<ReviewComment[]>>,
+) => {
+    if (comment.id) {
+        try {
+            const service = await GetCommentService()
+            const newComment = { ...reviewComments.find((c) => c.id === comment.id) }
```
Comment (taustad) [inline src/Components/SideSheet/Comments/Components/RenderComment.tsx:59]: Find a better name? Also move it to inside the if-statement, unless it is needed outside.

File: src/api/BaseService.ts (hunk 3 above)
Comment (fredrikleknes) [reply src/api/BaseService.ts]: its already removed

**Review by taustad — APPROVED** [review]

**Comment by fredrikleknes** [comment]
AB#123431

**Comment by fredrikleknes** [comment]
AB#123793

**Pull request merged**